saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/web/Saml2AuthenticationTokenConverter.java

```diff
@@ -86,16 +88,26 @@ public void setAuthenticationRequestRepository(
 		this.authenticationRequestRepository = authenticationRequestRepository;
 	}
 
+	/**
+	 * Use the given {@code shouldInflate} to inflate request.
+	 * @param shouldInflate the {@code shouldInflate} to use
+	 * @since 7.0
+	 */
+	public void setShouldInflateResponse(boolean shouldInflate) {
+		this.shouldInflate = shouldInflate;
+	}
+
 	private String decode(HttpServletRequest request) {
+		// prevent to break passivity in Saml2LoginBeanDefinitionParserTests
+		if (this.shouldInflate == null) {
+			this.shouldInflate = HttpMethod.GET.matches(request.getMethod());
```

Comment:
This is not thread-safe since multiple requests will use the same instance of this class. Please approach this in a way that does not involve updating a member variable.